SmartDeviceLink-iOS/SmartDeviceLink/SDLProxy.m

```diff
@@ -502,6 +505,32 @@ - (void)handleSystemRequestLaunchApp:(SDLOnSystemRequest *)request {
     }
 }
 
+- (void)handleSystemRequestLockScreenIconURL:(SDLOnSystemRequest *)request {
+    NSURLSessionDownloadTask *task = nil;
+    task = [self downloadTaskForSystemRequestURLString:request.url completionHandler:^(NSURL *location, NSURLResponse *response, NSError *error) {
+        
+        if ([self.activeSystemRequestTasks containsObject:task]) {
+            [self.activeSystemRequestTasks removeObject:task];
+        }
+        
+        if (error != nil) {
+            NSString *logMessage = [NSString stringWithFormat:@"OnSystemRequest failure (HTTP response), download task failed: %@", error.localizedDescription];
+            [SDLDebugTool logInfo:logMessage withType:SDLDebugType_RPC toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+            return;
+        }
+        
+        UIImage *icon = [UIImage imageWithData: [NSData dataWithContentsOfURL:location]];
+        SEL selector = @selector(onReceivedLockScreenIcon:);
+        [self.proxyListeners enumerateObjectsUsingBlock:^(id listener, NSUInteger idx, BOOL *stop) {
```

Comment:
see how this is made more clear in this PR: https://github.com/smartdevicelink/sdl_ios/blob/16fa7399dc1fe5430b878ed1bb06cee6aec9ce9d/SmartDeviceLink-iOS/SmartDeviceLink/SDLProxy.m#L627-634